apps/api/src/coscrad-cli/seed-test-data-with-command.cli-command.e2e.spec.ts

```diff
@@ -0,0 +1,170 @@
+import { TestingModule } from '@nestjs/testing';
+import { CommandTestFactory } from 'nest-commander-testing';
+import { AppModule } from '../app/app.module';
+import createTestModule from '../app/controllers/__tests__/createTestModule';
+import { ID_MANAGER_TOKEN } from '../domain/interfaces/id-manager.interface';
+import { CreateSong } from '../domain/models/song/commands/create-song.command';
+import { AggregateType } from '../domain/types/AggregateType';
+import { MockIdManagementService } from '../lib/id-generation/mock-id-management.service';
+import { NotFound } from '../lib/types/not-found';
+import { REPOSITORY_PROVIDER_TOKEN } from '../persistence/constants/persistenceConstants';
+import { ArangoConnectionProvider } from '../persistence/database/arango-connection.provider';
+import { ArangoDatabaseProvider } from '../persistence/database/database.provider';
+import TestRepositoryProvider from '../persistence/repositories/__tests__/TestRepositoryProvider';
+import generateDatabaseNameForTestSuite from '../persistence/repositories/__tests__/generateDatabaseNameForTestSuite';
+import { ArangoIdRepository } from '../persistence/repositories/arango-id-repository';
+import { CoscradCliModule } from './coscrad-cli.module';
+import { COSCRAD_LOGGER_TOKEN } from './logging';
+import { buildMockLogger } from './logging/__tests__';
+
+const cliCommandName = `seed-test-data-with-command`;
+
+const assertErrorMessageLogged = (
+    textToInclude: string,
+    actualCalls: string[][],
+    callIndex: [number, number] = [1, 0]
+) => {
+    /**
+     * We do things this way so that the failing error message will show up in
+     * the console whenever the test fails. This is far more useful than seeing
+     * > > > expect(isMesageContained).toBe(true)
+     *
+     * > > > expected true, received false
+     */
+    const unacceptableFailureMessages = [actualCalls[callIndex[0]][callIndex[1]]].filter(
+        (message) => !message.includes(textToInclude)
+    );
+
+    expect(unacceptableFailureMessages).toEqual([]);
+};
+
+describe(`CLI Command: ${cliCommandName}`, () => {
+    let commandInstance: TestingModule;
+
+    let testRepositoryProvider: TestRepositoryProvider;
+
+    let databaseProvider: ArangoDatabaseProvider;
+
+    let mockIdManager;
+
+    const mockLogger = buildMockLogger({ isEnabled: true });
+
+    beforeAll(async () => {
+        const testAppModule = await createTestModule({
+            ARANGO_DB_NAME: generateDatabaseNameForTestSuite(),
+        });
+
+        const arangoConnectionProvider =
+            testAppModule.get<ArangoConnectionProvider>(ArangoConnectionProvider);
+
+        databaseProvider = new ArangoDatabaseProvider(arangoConnectionProvider);
+
+        testRepositoryProvider = new TestRepositoryProvider(databaseProvider);
+
+        const idManager = new MockIdManagementService(new ArangoIdRepository(databaseProvider));
+
+        mockIdManager = {
```

Comment:
This is probably what we want every time. Look into moving this logic out into a test utility as soon as we need it again.